admin/code/AddToCampaignHandler.php

```diff
@@ -49,91 +58,96 @@ public function __construct($editForm, $data) {
 	/**
 	 * Perform the action. Either returns a Form or performs the action, as per the class doc
 	 *
-	 * @return HTML|SS_HTTPResponse
+	 * @return DBHTMLText|SS_HTTPResponse
 	 */
 	public function handle() {
 		$object = $this->getObject($this->data['ID'], $this->data['ClassName']);
 
 		if (empty($this->data['Campaign'])) {
 			return $this->Form($object)->forTemplate();
-		}
-		else {
+		} else {
 			return $this->addToCampaign($object, $this->data['Campaign']);
 		}
 	}
 
 	/**
 	 * Get what ChangeSets are available for an item to be added to by this user
+	 *
 	 * @return ArrayList[ChangeSet]
 	 */
 	protected function getAvailableChangeSets() {
 		return ChangeSet::get()
 			->filter('State', ChangeSet::STATE_OPEN)
 			->filterByCallback(function($item) {
-				return ($item->canView());
+				/** @var ChangeSet $item */
+				return $item->canView();
 			});
 	}
 
 	/**
 	 * Safely get a DataObject from a client-supplied ID and ClassName, checking: argument
 	 * validity; existence; and canView permissions.
 	 *
-	 * @param $id int - The ID of the DataObject
-	 * @param $class - The Class of the DataObject
-	 * @return DataObject - The referenced DataObject
+	 * @param int $id The ID of the DataObject
+	 * @param string $class The Class of the DataObject
+	 * @return DataObject The referenced DataObject
 	 * @throws SS_HTTPResponse_Exception
 	 */
 	protected function getObject($id, $class) {
 		$id = (int)$id;
 		$class = ClassInfo::class_name($class);
 
 		if (!$class || !is_subclass_of($class, 'DataObject') || !Object::has_extension($class, 'Versioned')) {
-			return $this->editForm->httpError(400, _t(
+			$this->editForm->httpError(400, _t(
```

Comment:
httpError doesn't return anything, but it does always throw an exception. You can't return a function that has a void return type (it causes php style warnings).

It would be fine to `return null;` if you're concerned about future changes.